check/configuration/custom-linters/reference.md

```diff
@@ -0,0 +1,527 @@
+
+[//]: # (`cache_results` : boolean, indicates whether or not to support caching for this linter.)
+
+[//]: # (`disable_upstream` :boolean, indicates whether or not we support comparing against the upstream version of this file.  **should we expose this?**)
+
+### `symlinks`
+
+`symlinks`: a list of symlinks to be created when using sandboxing.
+
+### `environment`
+
+`environment`: a list of runtime variables used when running the linter.
+
+### `is_recommended`
+
+`is_recommended`: *boolean*. Indicating whether `trunk init` should try to enable this linter. 
+Prefer [`suggest_if`](#suggest_if)
+
+### `run_linter_from`
+
+`run_linter_from`: indicates whether to set current working directory to WORKSPACE root, or the 
+target files folder when run. Prefer `run_from` at the command level.
+
+`include_scanner_type`: which include scanner to use, if any.
+
+### `formatter`
+
+`formatter`: boolean. Indicates whether this is a formatter and should be included in `trunk fmt`
+
+### `allow_empty_files`
+
+`allow_empty_files`: boolean. Indicates to skip linting empty files for this linter.
+
+### `runtime`
+
+`runtime`: RuntimeType, Which runtime, if any, to require to be setup for this linter.
+
+### `package`
+
+`package`: string, What primary package to install, if using a package manager runtime
+
+### `extra_packages`
+
+`extra_packages`: list of strings, Extra packages to install, versions are optional
+
+### `download`
+
+`download`: *string*. The download url. You must provide either runtime + packages or download, not both. Using runtimes is preferred. [See Runtimes](../../advanced-setup/runtimes.md)
+
+`issue_url_format`: string, a format string that accepts issue codes for links to issues docs.
+
+### `run_from_root_target`
+
+`run_from_root_target`: *string*. Walk up to find this file to detect the run from directory. 
+Prefer `run_from` at the command level.
+
+### `hold_the_line`
+
+`hold_the_line`: *optional boolean*. Whether hold-the-line will be done for this linter or not.
+
+### `read_output_from`
+
+`read_output_from`,  Tell the parser where to expect output from for reading (stdout, stderr, 
+tmp file). [See Output Sources](output-types.md#output-sources)
+
+### `prepare_command`
+
+`prepare_command`, ex. `[tflint, --init]`
+
+[//]: # (`version`, **???** )
+
+### `known_good_version`
+
+`known_good_version`: The version `trunk init` inits with. If not present Trunk will query the 
+version from its runtime or use the default version from the download.
+
+### `version_command`
+
+`version_command`:  Version check commands
+
+### `enabled`
+
+`enabled`: *optional boolean*. Whether this linter is enabled.
+
+### `batch`
+
+`batch`: *optional boolean*. Combine multiple files into the same execution.
+
+### `run_when`
+
+`run_when`, Should this linter be run only in CI? (or locally, vscode, monitor)
+
+### `in_place`
+
+`in_place`: *optional boolean*. Applies to formatters - does the formatter rewrite the file as 
```

Comment:
We should only say that this tells trunk that a formatting command will result in the file being rewritten in place.